Provider/src/FirebirdSql.EntityFrameworkCore.Firebird/Internal/FbOptions.cs

```diff
@@ -0,0 +1,231 @@
+            _connectionSettings = new Lazy<FbConnectionSettings>(() =>
+                {
+                    if (_contextOptions.Connection != null)
+                        return FbConnectionSettings.GetSettings(_contextOptions.Connection);
+                    return FbConnectionSettings.GetSettings(_contextOptions.ConnectionString);
+                }); 
+        }
+
+        public virtual void Validate(IDbContextOptions options)
+        {
+           //Removed Add Future!
+        }
+
+        public virtual FbConnectionSettings ConnectionSettings => _connectionSettings.Value;
+
+        public virtual string GetCreateTable(ISqlGenerationHelper sqlGenerationHelper, string table, string schema)
+        {
+            if (_contextOptions.Connection != null)
+                return GetCreateTable(_contextOptions.Connection, sqlGenerationHelper, table, schema);
+            return GetCreateTable(_contextOptions.ConnectionString, sqlGenerationHelper, table, schema);
+        }
+
+        private static string GetCreateTable(string connectionString, ISqlGenerationHelper sqlGenerationHelper, string table, string schema)
+        {
+            using (var connection = new FbConnection(connectionString))
+            {
+                connection.Open();
+                return ExecuteCreateTable(connection, sqlGenerationHelper, table, schema);
+            }
+        }
+
+        private static string GetCreateTable(DbConnection connection, ISqlGenerationHelper sqlGenerationHelper, string table, string schema)
+        {
+            var open = false;
+            if (connection.State == ConnectionState.Closed)
+            {
+                connection.Open();
+                open = true;
+            }
+            try
+            {
+                return ExecuteCreateTable(connection, sqlGenerationHelper, table, schema);
+            }
+            finally
+            {
+                if (open)
+                    connection.Close();
+            }
+        }
+        /// <summary>
+        /// Implementation for Scaffolding
+        /// </summary>
+        /// <param name="connection"></param>
+        /// <param name="sqlGenerationHelper"></param>
+        /// <param name="table"></param>
+        /// <param name="schema"></param>
+        /// <returns></returns>
+        /// Credit: http://www.firebirdfaq.org/faq174/  
+        private static string ExecuteCreateTable(DbConnection connection, ISqlGenerationHelper sqlGenerationHelper, string table, string schema)
+        {
+            using (var command = connection.CreateCommand())
+            {
+                //List Schema Table
+                command.CommandText = $@"SELECT
+                    RF.RDB$RELATION_NAME,
+                    RF.RDB$FIELD_NAME FIELD_NAME,
+                    RF.RDB$FIELD_POSITION FIELD_POSITION,
+                    CASE F.RDB$FIELD_TYPE
+                    WHEN 7 THEN
+                        CASE F.RDB$FIELD_SUB_TYPE
```

Comment:
Thanks, really valuable

Reply:
About the type 37, don't you think that it could be a good idea to get the information about the fact that it's a computed field and not just a varchar ? Provide a clear indication that it's a read only field imo.

```
WHEN 37 THEN
                        IIF (COALESCE(f.RDB$COMPUTED_SOURCE,'')<>'',
                        'COMPUTED BY ' || CAST(f.RDB$COMPUTED_SOURCE AS VARCHAR(250)),
                        'VARCHAR(' || (TRUNC(F.RDB$FIELD_LENGTH / CH.RDB$BYTES_PER_CHARACTER)) || ')')
```

Reply:
For the moment i leave the type 261 as it in the query as the rest of the code is eventually not prepared to received a CLOB/BLOB return.

`WHEN 261 THEN 'BLOB SUB_TYPE ' || F.RDB$FIELD_SUB_TYPE`

Reply:
Computed column can be on any datatype, not only varchars. Thus the `COMPUTED BY` should be then everywhere. Not sure how much it matters for scaffolding though.